### PR TITLE
[18913] Expose PayloadPool in DDSDataWriter and DDSDataReader

### PR DESCRIPTION
**Pull request by JesusPoderoso**

### CODE REVIEW

File: docs/fastdds/dds_layer/publisher/dataWriter/createDataWriter.rst [hunk 1]
```diff
@@ -78,6 +78,24 @@ It is advisable to check that the returned value is a valid pointer.
...
    :dedent: 8
 
+.. _dds_layer_publisher_datawriter_with_payload_pool_creation:
+
+Creating a DataWriter with a custom PayloadPool
+-----------------------------------------------
+
+A custom :ref:`PayloadPool<rtps_layer_custom_payload_pool>` can be passed as an argument during the creation of a
+:ref:`dds_layer_publisher_dataWriter`.
+That would allow customizing the management of the information exchanged between DataWriters and DataReaders.
```
Comment (EduPonz) [inline docs/fastdds/dds_layer/publisher/dataWriter/createDataWriter.rst:88]:
```suggestion
This allows for customizing the management of the information exchanged between DataWriters and DataReaders.
```

File: code/DDSCodeTester.cpp [hunk 2]
```diff
@@ -2535,6 +2535,23 @@ void dds_dataWriter_examples()
         //!--
     }
 
+    {
+        //DDS_CREATE_PAYLOAD_POOL_DATAWRITER
+        // A DataWriterQos must be provided to the creation method
+        DataWriterQos qos;
+
+        // Create PayloadPool
+        std::shared_ptr<CustomPayloadPool> payload_pool = std::make_shared<CustomPayloadPool>();
```
Comment (EduPonz) [inline code/DDSCodeTester.cpp:2544]: I don't this this would compile since `CustomPayloadPool` is not defined as far as I can tell.

**Review by EduPonz — CHANGES_REQUESTED** [review]
Maybe it's a good idea to reopen this PR from a branch named the same as the PR in Fast DDS so that both CI run correctly

**Comment by JesusPoderoso** [comment]
Closing this PR, this changes have been moved to #514

**Pull request closed without merging**